packages/background-charm-service/integration/counter.test.ts

```diff
@@ -0,0 +1,112 @@
+import { env } from "@commontools/integration";
+import { sleep } from "@commontools/utils/sleep";
+import {
+  registerCharm,
+  ShellIntegration,
+} from "@commontools/integration/shell-utils";
+import { describe, it } from "@std/testing/bdd";
+import { join } from "@std/path";
+import { assert, assertEquals } from "@std/assert";
+
+const { API_URL, FRONTEND_URL } = env;
+
+describe("background charm counter tests", () => {
+  const shell = new ShellIntegration();
+  shell.bindLifecycle();
+
+  it("can register and interact with background counter charm", async () => {
+    // FIXME(ja): currently bg process doesn't receive updates of bgCharms,
+    // and so we need to start it after we register :(  We should start it here
+    // this.charmCell.sink only seems to trigger when the service.ts starts -
+    // restarting is currently the only way to get the charmCell to update
+
+    const { page, identity } = shell.get();
+    const spaceName = globalThis.crypto.randomUUID();
+
+    const charmId = await registerCharm({
+      spaceName: spaceName,
+      apiUrl: new URL(API_URL),
+      identity: identity,
+      source: await Deno.readTextFile(
+        join(
+          import.meta.dirname!,
+          "..",
+          "..",
+          "..",
+          "recipes",
+          "bgCounter.tsx",
+        ),
+      ),
+    });
+
+    // TODO(js): Remove /shell when no longer prefixed
+    await page.goto(`${FRONTEND_URL}shell/${spaceName}/${charmId}`);
```

Comment:
Will need to remove `shell` if https://github.com/commontoolsinc/labs/pull/1573 lands first